### DISCUSSION TITLE
Animated Bitmap Help Node-red

### DISCUSSION THREAD

**Discussion by keshin88** [pixelit-project/PixelIt#191] [Q&A] [answered]
hey there i have successfully made and implemented pixelit into node-red and looked at the example flow. 
i have successfully made and changed and can use text scrolling and bitmap images, i however have an issue i cannot work out.

Issue:
The Animated Bitmap i cannot work out how to use in a 8x32 animated bitmap. it tries to display it all solely on 8x8. 

Possibly related:
in the pixelit dashboard under gallery, i can succesfully view 8x32 static bitmap, but if i try to display an animated one it will not send or update.  if i copy the code and test it in the test area i get the animated image but confined to the 8x8 grid instead of the 32x8

![flow](https://user-images.githubusercontent.com/68045979/187013370-39423834-2f78-4e20-b6ed-5033f4334bcb.png)

my exported flow Below

`[{"id":"1e0df6f10a2878df","type":"Core","z":"e75e42047510d888","name":"Core","ip":"192.168.1.31","masterTopic":"","x":730,"y":100,"wires":[[]]},{"id":"a1f267fd8ec13237","type":"Alert Screen","z":"e75e42047510d888","name":"Alert Screen","x":590,"y":100,"wires":[["1e0df6f10a2878df"]]},{"id":"bc0e2ea49924f309","type":"Bitmap Animation","z":"e75e42047510d888","name":"Bitmap Animation","indata":"[0,0,0,63488,63488,63488,0,0,63488,0,0,0,0,0,63488,63488,63488,0,0,63488,63488,63488,63488,0,0,63488,0,0,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,63488,0,63488,63488,0,63488,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,63488,0,63488,0,63488,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,63488,0,63488,0,0,0,63488,0,0,0,0,63488,63488,63488,63488,63488,0,63488,0,0,0,0,63488,63488,63488,63488,63488,0,63488,63488,63488,63488,0,0,63488,0,0,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,63488,0,0,0,63488,0,0,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,0,0,63488,0,0,63488,0,0,0,63488,0,0,0,0,63488,0,0,0,63488,0,63488,63488,63488,63488,0,63488,0,0,0,63488,0,63488,0,0,0,63488,0,63488,0,0,0,63488,0,0],[31,31,31,63488,63488,63488,31,31,63488,31,31,31,31,31,63488,63488,63488,31,31,63488,63488,63488,63488,31,31,63488,31,31,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,63488,31,63488,63488,31,63488,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,63488,31,63488,31,63488,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,63488,31,63488,31,31,31,63488,31,31,31,31,63488,63488,63488,63488,63488,31,63488,31,31,31,31,63488,63488,63488,63488,63488,31,63488,63488,63488,63488,31,31,63488,31,31,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,63488,31,31,31,63488,31,31,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,31,31,63488,31,31,63488,31,31,31,63488,31,31,31,31,63488,31,31,31,63488,31,63488,63488,63488,63488,31,63488,31,31,31,63488,31,63488,31,31,31,63488,31,63488,31,31,31,63488,31,31]","inanimationDelay":"100","inrubberbanding":"false","inlimitLoops":"0","x":410,"y":100,"wires":[["a1f267fd8ec13237"]]},{"id":"0fa3453230a4c6d7","type":"Screen Init","z":"e75e42047510d888","name":"Screen Init","inscreenName":"Postman","induration":"25","x":230,"y":100,"wires":[["bc0e2ea49924f309"]]},{"id":"98daa98d3e15fb95","type":"inject","z":"e75e42047510d888","name":"test","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":true,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":90,"y":100,"wires":[["0fa3453230a4c6d7"]]}]`

**Comment by o0shojo0o** [comment, marked as answer]
The PixelIt supports only 8x8 Animated Bitmaps due to lack of memory. 
https://pixelit-project.github.io/api.html#bitmap-animation

**Reply by keshin88** [reply]
oh ok, thank you, i thought 32x8 must have been available as in the gallery it shows the animated alarm 32x8. my mistake thank you for clearing that up

**Reply by o0shojo0o** [reply]
Thanks for the tip, I had not noticed that users have tried to create 8x32 animations I have just removed them from the database. 
That here no further confusion can arise.

**Reply by keshin88** [reply]
thank you, is there plans for it in the future? i know the build asked for a d1mini however i have done mine with the esp8266 nodemcu.
forgive me im not clued on with progamming but if such a feature would be possible in the future that would be awesome. 
this is a great project, also i may have missed it but is it possible to expand and have for eg 4 32x8 arrays connected as a larger panel?

**Reply by o0shojo0o** [reply]
> thank you, is there plans for it in the future?

No, I have already made several attempts but I have always come up against the lack of RAM of the ESP.

> this is a great project, also i may have missed it but is it possible to expand and have for eg 4 32x8 arrays connected as a larger panel?

No, this is not possible because the functions and their logics are based on the number of pixels 8x32.
